hackathon_site/event/api_views.py

```diff
@@ -109,6 +110,26 @@ def post(self, request, *args, **kwargs):
         return Response(data=response_data, status=status.HTTP_200_OK,)
 
 
+class TeamIncidentListView(
+    mixins.ListModelMixin, mixins.CreateModelMixin, generics.GenericAPIView
+):
+    queryset = Incident.objects.all()
+
+    serializer_class = IncidentCreateSerializer
+    permission_classes = [UserHasProfile]
+
+    def perform_create(self, serializer):
+        incident_team = serializer.validated_data["order_item"].order.team
+        user_team = self.request.user.profile.team
+
+        if incident_team != user_team:
+            raise PermissionDenied("Can only post incidents for your own team.")
```

Comment:
Nit:
```suggestion
            raise PermissionDenied("Can only create incidents for your own team.")
```

Just making the message more user-friendly if we expose these messages somewhere (not that our frontend will let someone create an incident for another team)